Modify sponsorship expiration

teovin opened the pull request:
This PR adds the ability to revise a sponsored user's affiliation expiration date through the manage sponsored user view. 

Admin users can see all sponsorships, while registrar users can only see the sponsorships that are affiliated with their own registrar.

Here is the before and after of how the `manage/sponsored-users/<user_id>` page will look:

**Before:**

![image](https://github.com/user-attachments/assets/32867c08-a8d2-47dd-8f8e-c75cc62cfe4b)

**After:**

![Screenshot 2025-04-16 at 8 34 53 AM](https://github.com/user-attachments/assets/468eb7cc-eead-4d14-96a8-9e34ae68f876)

![Screenshot 2025-04-16 at 8 35 02 AM](https://github.com/user-attachments/assets/02c73f40-9844-4c3c-9fc3-a0f5621a43b4)

Notice that I added a new button next to the `Deactivate sponsorship` button. They both have the same initial orange bordered styling, but the `Update sponsorship` button turns green on hover and focus events while the deactivate button turns red. 

Another change I made in this view is that previously, we weren't displaying affiliation information for permanent affiliations, and with this change, I added a permanent affiliation text. I thought this would make it clearer to the admins/registrars that permanent affiliations are also modifiable. I also removed the expiration info from inactive sponsorships (which I am fine with adding back; I thought this made more sense).

**Here is how the form to update the date looks like:**

![Screenshot 2025-04-11 at 9 42 25 AM](https://github.com/user-attachments/assets/c3c5ac3d-1661-4424-9716-b470399a31c4)

If the user checks the `Sponsor permanently` box, the date selection input will be disabled. Only when the user de-selects the checkbox will the date selection be re-enabled:

![image](https://github.com/user-attachments/assets/9b36a905-a758-4881-a217-2fdbd4a8ecae)

Just a note that I haven't run this by Clare yet; I am planning to when she returns. So there might be changes to wording, styling, and overall behavior.

teovin committed:
first draft of changes

teovin committed:
update wording

teovin committed:
indentation

teovin committed:
remove bold

teovin committed:
use absolute url

rebeccacremona reviewed (approved):
Huzzah.

LGTM! I personally like that you removed the expiration date from inactive sponsorships; agreed that looks cleaner. One suggestion for wording: it might be clearer to say, "Permanent affiliation" rather than, "Expiration: Permanent affiliation" but you mentioned you were going to talk to Clare about labels anyway 🙂.

How would you feel about adding a couple tests?

teovin commented:
Thanks Becky, yep, will check with Clare as well and update labels.

And tests, I guess I can 🤣

teovin committed:
add tests

teovin committed:
test checking the date change for funsies

teovin committed:
typo

teovin committed:
remove expiration label if permanent affiliation

teovin commented:
@rebeccacremona, regarding your comment about the expiration label for the permanent affiliations, I removed that and updated the screenshots in the PR description to reflect that. I will go ahead and close this PR for now, and if there is any additional feedback on the labels, etc, from the rest of the team, I can open another PR then.
